A working api authorization

Thanerik opened the pull request:
It works by if you logs in, it creates an access token, and set it in your cookie as an encrypted httpOnly cookie (meaning that no javascript have access to it). The cookie is thereafter "converted" into a header authorization bearer token by each request, making it a valid backend authorization process, and thus allowing all the requests to be validated.
It means now you can either authorize yourself using the cookie set when you logs in using frontend login formula, or by request a access token using the normal way (by api with credentials), and from that send the token with by each request, and thus authorize yourself.
All graphQL calls in frontend works. All GraphQL calls from external sources (as Playground) works with header access token set. You can chose in Postman or other softwares to either send the encrypted cookie in request, or use bearer token.
We must remember to set the COOKIE_SECURE_HTTP env variable to true in production, thus enabling it to force HTTPS usage in the cookie.

Thanerik committed:
Improvements...

Thanerik committed:
Merge branch 'develop' of https://github.com/Wign/Wign-design-v2 into feature/backToFront

Thanerik committed:
Now with a working api authorization.
It works by if you logs in, it creates a access token, and set it in your cookie as a encrypted httpOnly cookie (meaning that no javascript have access to it). The cookie is thereafter "converted" into a header authorization bearer token by each request, making it a valid backend Authorization process, and thus allowing all the requests to be validated.
It means now you can either authorize yourself using the cookie set when you logs in using frontend login formula, or by request a access token using the normal way (by api with credentials), and from that send the token with by each request, and thus authorize yourself.
All graphQL calls in frontend works. All GraphQL calls from external sources (as Playground) works with header access token set. You can chose in Postman or other softwares to either send the encrypted cookie in request, or use bearer token.
We must remember to set the COOKIE_SECURE_HTTP env variable to true in production, thus enabling it to force HTTPS usage in the cookie.

Thanerik commented:
It fixes issue #30

Trudellik commented on config/auth.php:
```diff
@@ -31,7 +31,7 @@
     | mechanisms used by this application to persist your user's data.
     |
-    | Supported: "session", "token"
+    | Supported: "session", "token", 'passport'
```
Should it be "passport" instead of 'passport'?

Thanerik replied:
Yeah - just a commented text, so it make no difference :p

Trudellik reviewed (approved):
Jeg har ikke meget at kommentere udover at ville vide om er det også sikret at tokenen bliver straks ugyldigt så snart man logger af? Altså at kalde toggleLike uden om fronten ikke vil virke efter man har logget af.

Thanerik commented:
> ville vide om er det også sikret at tokenen bliver straks ugyldigt så snart man logger af? Altså at kalde toggleLike uden om fronten ikke vil virke efter man har logget af.

Du har ret i at der ikke er blevet taget højde for at ugyldiggøre tokens når man er logget af. Det skal vi også have på plads, så det implementerer jeg, før jeg smider denne ændring op. Det er især at token ikke invalideres når man er logget af. Tokens har en vis levetid, som selv udløber (efter 24 timer cirka). Men hvis man fysisk har klikket på "log af", så skal det sgu virke

Thanerik committed:
Now with a working api authorization including logout.
It works by if you logs in, it creates a Javascript Web Token and set it in your cookie as a encrypted httpOnly cookie (meaning that no javascript have access to it). The cookie is thereafter compared with the csrf token extracted from the HEAD section of the website and sent in as a header by Apollo. It makes it a valid backend Authorization process, and thus allowing all the requests to be validated. The JWT token has an expiration time of 1 hour. And each time you logs off it deletes the cookie, making it impossible for you to fiddle with login and authorization.
It means now you either authorize yourself by using the JWT cookie set when you logs in in frontend, or by request a access token using the normal api way (by api with credentials). If it's JWT token, it only allows you to make requests in the active session you are in. In the api way you gain a access token you can use as much as you want in it's lifetime (one year) to send with all your requests, and thus authorize yourself.
All graphQL calls in frontend works. All GraphQL calls from external sources (as Playground) works with header access token set. In Postman or other softwares, you can chose to either set the encrypted JWT token in cookie as `laravel_token` AND set the request header with `X-CSRF-TOKEN` with unencrypted csrf-token as found in HEAD section in HTML - OR just use the bearer authorization.
We must remember to set the `SESSION_SECURE_COOKIE` env variable to true in production (defaults to false), thus enabling it to force HTTPS usage in the cookie.

Thanerik committed:
Fixed StyleCI

Thanerik committed:
Merge branch 'develop' into feature/authorization
